test/apivalidations/flavor_test.go

```diff
@@ -86,4 +104,114 @@ var _ = Describe("ORC Flavor API validations", func() {
+		Expect(k8sClient.Create(ctx, flavor)).To(Succeed())
+		Expect(flavor.Spec.ManagementPolicy).To(Equal(orcv1alpha1.ManagementPolicyManaged))
+	})
+
+	It("should require import for unmanaged", func(ctx context.Context) {
+		flavor := flavorStub(namespace)
+		patch := baseFlavorPatch(flavor)
+		patch.Spec.WithManagementPolicy(orcv1alpha1.ManagementPolicyUnmanaged)
+		Expect(applyObj(ctx, flavor, patch)).NotTo(Succeed())
+
+		patch.Spec.WithImport(testFlavorImport())
+		Expect(applyObj(ctx, flavor, patch)).To(Succeed())
+	})
+
+	It("should not permit unmanaged with resource", func(ctx context.Context) {
+		flavor := flavorStub(namespace)
+		patch := baseFlavorPatch(flavor)
+		patch.Spec.
+			WithManagementPolicy(orcv1alpha1.ManagementPolicyUnmanaged).
+			WithImport(testFlavorImport()).
+			WithResource(testFlavorResource())
+	})
+
+	It("should not permit empty import", func(ctx context.Context) {
+		flavor := flavorStub(namespace)
+		patch := baseFlavorPatch(flavor)
+		patch.Spec.
+			WithManagementPolicy(orcv1alpha1.ManagementPolicyUnmanaged).
+			WithImport(applyconfigv1alpha1.FlavorImport())
+		Expect(applyObj(ctx, flavor, patch)).NotTo(Succeed())
+	})
+
+	It("should not permit empty import filter", func(ctx context.Context) {
+		flavor := flavorStub(namespace)
+		patch := baseFlavorPatch(flavor)
+		patch.Spec.
+			WithManagementPolicy(orcv1alpha1.ManagementPolicyUnmanaged).
+			WithImport(applyconfigv1alpha1.FlavorImport().
+				WithFilter(applyconfigv1alpha1.FlavorFilter()))
+		Expect(applyObj(ctx, flavor, patch)).NotTo(Succeed())
+	})
+
+	It("should permit import filter with name", func(ctx context.Context) {
+		flavor := flavorStub(namespace)
+		patch := baseFlavorPatch(flavor)
+		patch.Spec.
+			WithManagementPolicy(orcv1alpha1.ManagementPolicyUnmanaged).
+			WithImport(applyconfigv1alpha1.FlavorImport().
+				WithFilter(applyconfigv1alpha1.FlavorFilter().WithName("foo")))
+		Expect(applyObj(ctx, flavor, patch)).To(Succeed())
+	})
+
+	It("should require resource for managed", func(ctx context.Context) {
+		flavor := flavorStub(namespace)
+		patch := baseFlavorPatch(flavor)
+		patch.Spec.WithManagementPolicy(orcv1alpha1.ManagementPolicyManaged)
+		Expect(applyObj(ctx, flavor, patch)).NotTo(Succeed())
+
+		patch.Spec.WithResource(testFlavorResource())
+		Expect(applyObj(ctx, flavor, patch)).To(Succeed())
+	})
+
+	It("should not permit managed with import", func(ctx context.Context) {
+		flavor := flavorStub(namespace)
+		patch := baseFlavorPatch(flavor)
+		patch.Spec.
+			WithImport(testFlavorImport()).
+			WithManagementPolicy(orcv1alpha1.ManagementPolicyManaged).
+			WithResource(testFlavorResource())
+		Expect(applyObj(ctx, flavor, patch)).NotTo(Succeed())
+	})
+
+	It("should require content when not importing", func(ctx context.Context) {
+		flavor := flavorStub(namespace)
+		patch := baseFlavorPatch(flavor)
+		patch.Spec.WithResource(applyconfigv1alpha1.FlavorResourceSpec())
+		Expect(applyObj(ctx, flavor, patch)).NotTo(Succeed())
+	})
```

Comment:
I think this only makes sense for Image. `Content` is a property of ImageResourceSpec.